lib/samples/find_address_with_reverse_geocode/find_address_with_reverse_geocode_sample.dart

```diff
@@ -47,158 +51,78 @@ class _FindAddressWithReverseGeocodeSampleState
-            color: Colors.white,
-            child: TextField(
-              decoration: const InputDecoration(
-                hintText: 'Search...',
-              ),
-              onSubmitted: onSearchSubmitted,
-            ),
-          ),
-        ),
-        IconButton(
-          icon: const Icon(
-            Icons.list,
-            color: Colors.white,
-          ),
-          onPressed: () => _scaffoldStateKey.currentState!.openEndDrawer(),
-        ),
-      ],
-    );
-  }
-
   void onMapViewReady() async {
-    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISImagery);
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISTopographic);
 
     final image = await ArcGISImage.fromAsset('assets/pin_circle_red.png');
     final pictureMarkerSymbol = PictureMarkerSymbol.withImage(image);
     pictureMarkerSymbol.width = 35;
     pictureMarkerSymbol.height = 35;
     pictureMarkerSymbol.offsetY = pictureMarkerSymbol.height / 2;
+
     _graphicsOverlay.renderer = SimpleRenderer(symbol: pictureMarkerSymbol);
-    _graphicsOverlay.graphics.add(_graphic);
     _mapViewController.graphicsOverlays.add(_graphicsOverlay);
 
     _mapViewController.arcGISMap = map;
-
-    _geocodeParameters.minScore = 75;
-    _geocodeParameters.resultAttributeNames
-        .addAll(['Place_addr', 'Match_addr']);
+    _mapViewController.setViewpoint(_initialViewpoint);
 
     await Future.wait([
       _worldLocatorTask.load(),
-      _sanDiegoLocatorTask.load(),
     ]);
 
     setState(() => _ready = true);
   }
 
   void onTap(Offset localPosition) async {
-    final identifyGraphicsOverlayResult =
-        await _mapViewController.identifyGraphicsOverlay(
-      _graphicsOverlay,
-      screenPoint: localPosition,
-      tolerance: 5,
-    );
+    if (_graphicsOverlay.graphics.isNotEmpty) _graphicsOverlay.graphics.clear();
 
-    if (identifyGraphicsOverlayResult.graphics.isEmpty) return;
+    final mapTapPoint =
+        _mapViewController.screenToLocation(screen: localPosition);
+    if (mapTapPoint == null) return;
 
-    if (mounted) {
-      final graphic = identifyGraphicsOverlayResult.graphics.first;
-      final center = graphic.geometry!.extent.center;
-      final x = center.x.toStringAsFixed(3);
-      final y = center.y.toStringAsFixed(3);
-      final matchAddr = graphic.attributes['Match_addr'];
+    final normalizedTapPoint =
+        GeometryEngine.normalizeCentralMeridian(geometry: mapTapPoint);
+    displayTappedPoint(normalizedTapPoint!);
+
+    final reverseGeocodeParameters = ReverseGeocodeParameters();
+    reverseGeocodeParameters.maxResults = 1;
+    final reverseGeocodeResult = await _locatorTask.reverseGeocode(
+      location: normalizedTapPoint as ArcGISPoint,
+      parameters: reverseGeocodeParameters,
+    );
+    if (reverseGeocodeResult.isEmpty) return;
+
+    try {
+      final firstResult = reverseGeocodeResult.first;
+      final address = firstResult.attributes['LongLabel'] as String;
```

Comment:
Why is this line inside the `try` block? Is it possible to get a `StateError` out of it? If not, I would think it should go before the `try` block.

Reply:
I am not sure if it was StateError, but this line was throwing some error when I indicated a wrong attribute key. But since I know "LongLabel" is the right key, it shouldn't be throwing any errors - I will take these two lines out!